source/common/formatter/substitution_formatter.cc

```diff
@@ -953,6 +954,15 @@ StreamInfoFormatter::StreamInfoFormatter(const std::string& field_name) {
           }
           return absl::nullopt;
         });
+  } else if (field_name == "VIRTUAL_CLUSTER_NAME") {
+    field_extractor_ = std::make_unique<StreamInfoStringFieldExtractor>(
+        [](const StreamInfo::StreamInfo& stream_info) {
+          absl::optional<std::string> virtual_cluster_name = stream_info.virtualClusterName();
+          if (virtual_cluster_name.has_value()) {
+            return virtual_cluster_name.value();
+          }
+          return DefaultUnspecifiedValueString;
```

Comment:
I believe FieldExtractor's return value is an optional string and we should return `absl::null_opt` here. In fact, I think the entire function body can just be `return stream_info.virtualClusterName()`.

This allows us to distinguish whether the value is set or not (e.g. `absl::null_opt`). The decision whether to render a hyphen for unset values is left to code that understands the context and whether we need a `ProtobufWkt::Value` (which can encode a null value) or a string that defaults to `"-"`.

Reply:
@zuercher Thanks for taking a look. I changed it to return an optional as you suggested.